docs/internals.rst

```diff
@@ -119,48 +119,100 @@ Provider Configurations
 Request Caching
 ```````````````
 
+.. important::
+    Familiarize yourself with the validation logic for request caching before
+    enabling it. Since this feature often requires making additional requests under the
+    hood to try to guarantee the validity of the data, it may create unnecessary
+    overhead for your use case. Validation can be turned off by setting the
+    ``request_cache_validation_threshold`` option to ``None``, caching all allowed
+    requests, or configured for adjusting performance to your needs.
+
+
 Request caching can be configured at the provider level via the following configuration
 options on the provider instance:
 
 - ``cache_allowed_requests: bool = False``
-- ``cacheable_requests: Set[RPCEndpoint] = CACHEABLE_REQUESTS``
-- ``request_cache_validation_threshold: RequestCacheValidationThreshold = RequestCacheValidationThreshold.FINALIZED``
+- ``cacheable_requests: Optional[Set[RPCEndpoint]]``
+- ``request_cache_validation_threshold: Optional[Union[RequestCacheValidationThreshold, int]]``
 
 For requests that don't rely on block data (e.g., ``eth_chainId``), enabling request
 caching by setting the ``cache_allowed_requests`` option to ``True`` will cache all
 responses. This is safe to do.
 
 However, for requests that rely on block data (e.g., ``eth_getBlockByNumber``), it is
 not safe to always cache their responses because block data can change - during a
-chain reorganization, for example. The ``request_cache_validation_threshold`` option
-allows configuring a safe threshold for caching responses that depend on block data. By
-default, the ``finalized`` block number is used as the validation threshold, meaning
-that a request's response will be cached if the block number it relies on is less than
-or equal to the ``finalized`` block number. If the block number exceeds the
-``finalized`` block number, the response won't be cached.
+chain reorganization or while finality has not been reached, for example. The
+``request_cache_validation_threshold`` option allows configuring a safe threshold for
+caching responses that depend on block data. By default, this option is configured
+to internal values deemed "safe" for the chain id you are connected to. If you are
+connected to mainnet Ethereum, this value is set to the ``finalized`` block number.
+If you are connected to another chain, this value is set to a time internal in seconds,
```

Comment:
did you mean "is set internally to a time in seconds" or something like that?

Reply:
Womp... "time interval". Will fix